sqlglotrs/src/settings.rs

```diff
@@ -0,0 +1,462 @@
+use crate::token::TokenType;
+use pyo3::prelude::*;
+use std::collections::{HashMap, HashSet};
+use std::str::FromStr;
+
+#[derive(Clone, Debug)]
+#[pyclass]
+pub struct TokenizerSettings {
+    pub white_space: HashMap<char, TokenType>,
+    pub single_tokens: HashMap<char, TokenType>,
+    pub keywords: HashMap<String, TokenType>,
+    pub numeric_literals: HashMap<String, String>,
+    pub identifiers: HashMap<char, char>,
+    pub identifier_escapes: HashSet<char>,
+    pub string_escapes: HashSet<char>,
+    pub escape_sequences: HashMap<String, String>,
+    pub quotes: HashMap<String, String>,
+    pub format_strings: HashMap<String, (String, TokenType)>,
+    pub has_bit_strings: bool,
+    pub has_hex_strings: bool,
+    pub comments: HashMap<String, Option<String>>,
+    pub var_single_tokens: HashSet<char>,
+    pub commands: HashSet<TokenType>,
+    pub command_prefix_tokens: HashSet<TokenType>,
+    pub identifiers_can_start_with_digit: bool,
+}
+
+impl TokenizerSettings {
+    pub fn default() -> TokenizerSettings {
+        TokenizerSettings {
+            white_space: HashMap::from([
```

Comment:
why do we need to copy this over? can python hand it over? i don't want to have to manually update rust everytime i add a token to python